sdks/go/pkg/beam/transforms/sql/sql_test.go

```diff
@@ -0,0 +1,105 @@
+package sql
+
+import (
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/transforms/sql/sqlx"
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+func TestOptions_Add(t *testing.T) {
+	o := options{}
+	str := "this is a string"
+	bytes := []byte{1, 2, 3, 4}
+	testOpt := sqlx.Option{
+		Urn:     str,
+		Payload: bytes,
+	}
+	tests := []struct {
+		opt sqlx.Option
+	}{
+		{
+			opt: testOpt,
+		},
+	}
+
+	for _, test := range tests {
+		o.Add(test.opt)
+		if o.customs == nil || !reflect.DeepEqual(o.customs[len(o.customs)-1], test.opt) {
+			t.Errorf("Add failed.")
+		}
+	}
+}
+
+func TestInput(t *testing.T) {
+	name1 := "this is a string"
+	collection1 := beam.PCollection{}
+	tests := []struct {
+		name string
+		in   beam.PCollection
+	}{
+		{
+			name: name1,
+			in:   collection1,
+		},
+	}
+
+	o := &options{inputs: make(map[string]beam.PCollection)}
+	for _, test := range tests {
+		option := Input(test.name, test.in)
+		if option == nil {
+			t.Errorf("%v return a nil value, which is not expected", runtime.FuncForPC(reflect.ValueOf(TestInput).Pointer()).Name())
```

Comment:
While it's *neat* to us the runtime function magically find the test name, it's not useful in this context.

As written this will produce a test error string of
`TestInput return a nil value, which is not expected`, which isn't useful, since one already had to type "TestInput" to get it correct. 

Also, it's referring to the test case function itself `TestInput`, not the function under test `Input`. Simply use the text directly instead of abstracting it unnecessarily.

If you did intend to get the test name itself, use `t.Name()` instead. 
https://pkg.go.dev/testing#T.Name

Reply:
Got it, I will simply use the text directly, followed by [https://google.github.io/styleguide/go/decisions#identify-the-function](https://google.github.io/styleguide/go/decisions#identify-the-function)